lib/LaTeXML/Core/Mouth/file.pm

```diff
@@ -63,11 +65,17 @@ sub getNextLine {
       close($fh); $$self{IN} = undef;
       return; }
     else {
+      if (!$$self{encoding}) {
+        my $detected = detect($line);
+        if ($detected && $detected eq 'UTF-8') {
+          $$self{encoding} = 'UTF-8';
+        }
+      }
       push(@{ $$self{buffer} }, LaTeXML::Core::Mouth::splitLines($line)); } }
 
   my $line = shift(@{ $$self{buffer} });
   if (defined $line) {
-    if (my $encoding = $STATE->lookupValue('PERL_INPUT_ENCODING')) {
+    if (my $encoding = $$self{encoding}) {
```

Comment:
I also snuck in an optimization here - we were calling `lookupValue` on every line, but we can really just call it once when the Mouth object gets created.